xmtp_proto/src/api_client.rs

```diff
@@ -368,29 +368,22 @@ where
         (**self).stats()
     }
 }
-#[cfg(not(target_arch = "wasm32"))]
+
 #[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
 pub trait XmtpMlsStreams {
+    #[cfg(not(target_arch = "wasm32"))]
     type GroupMessageStream: Stream<Item = Result<GroupMessage, Self::Error>> + Send;
 
-    type WelcomeMessageStream: Stream<Item = Result<WelcomeMessage, Self::Error>> + Send;
-    type Error: RetryableError + Send + Sync + 'static;
+    #[cfg(target_arch = "wasm32")]
+    type GroupMessageStream: Stream<Item = Result<GroupMessage, Self::Error>>;
```

Comment:
Instead of adding these extra cfg directives everywhere it would be better to add a trait like `trait MaybeSend` that is configured to optionally depend on `Send` and do a blanket impl for all types that meet the constraint then constrain things here to `+ MaybeSend` which should do the right thing and get rid of cfg like this.